torch/_inductor/codecache.py

```diff
@@ -230,15 +231,15 @@ def write(source_code, ext, extra=""):
 
 
 def write_atomic(path: str, source_code: str):
-    # Write into tempdir first to avoid conflicts between threads
+    # Write into temporary file first to avoid conflicts between threads
     # Avoid using a named temporary file, as those have restricted permissions
+    salt = f"{os.getpid()}.{threading.get_ident()}"
     path = pathlib.Path(path)
-    with tempfile.TemporaryDirectory(dir=path.parent) as tempdir:
-        tmp_path = pathlib.Path(tempdir) / path.name
-        with tmp_path.open("w") as f:
-            f.write(source_code)
+    tmp_path = path.parent / f".{path.name}.{salt}.tmp"
```

Comment:
Don't include `path.name` here, we are getting pretty close to the filename length limit on some older file system types.